File: galaxy-web-app/galaxy-oj/src/main/java/io/charlie/web/oj/modular/data/judgecase/entity/DataJudgeCase.java
package io.charlie.web.oj.modular.data.judgecase.entity;

import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableName;
import com.baomidou.mybatisplus.annotation.TableName;
import io.charlie.galaxy.pojo.CommonEntity;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.io.Serial;
import java.util.Date;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.dromara.core.trans.anno.Trans;
import org.dromara.core.trans.constant.TransType;

/**
* @author Charlie Zhang
* @version v1.0
* @date 2025-10-26
* @description 判题结果用例表
*/
@EqualsAndHashCode(callSuper = true)
@Data
@TableName("data_judge_case")
@Schema(name = "DataJudgeCase", description = "判题结果用例表")
public class DataJudgeCase extends CommonEntity {
    @Serial
    @TableField(exist = false)
    private static final long serialVersionUID = 1L;

    @Schema(description = "主键")
    private String id;

    @Schema(description = "提交ID")
    private String submitId;
    @Schema(description = "用例标识")
    private String caseSign;
    @Schema(description = "输入数据")
    private String inputData;

    @Schema(description = "输出数据")
    private String outputData;

    @Schema(description = "期望输出")
    private String expectedOutput;

    @Schema(description = "输入文件路径")
    private String inputFilePath;

    @Schema(description = "输入文件大小")
    private Long inputFileSize;

    @Schema(description = "输出文件路径")
    private String outputFilePath;

    @Schema(description = "输出文件大小")
    private Long outputFileSize;

    @Schema(description = "最大耗时")
    private BigDecimal maxTime;

    @Schema(description = "最大内存使用")
    private BigDecimal maxMemory;

    @Schema(description = "是否样例判题")
    @Trans(type = TransType.DICTIONARY, key = "YES_NO")
    private Boolean isSample;

    @Schema(description = "分值")
    private BigDecimal score;

    @Schema(description = "执行状态")
    private String status;

    @Schema(description = "信息")
    private String message;

    @Schema(description = "退出码")
    private Integer exitCode;
}

File: judge-service-1/internal/database/model/data_judge_case.go
package model

import (
	"time"
)

// 判题结果用例表
type DataJudgeCase struct {
	ID             string     `gorm:"column:id;primaryKey;type:varchar(32)"`
	SubmitID       string     `gorm:"column:submit_id;type:varchar(32);not null"`
	CaseSign       string     `gorm:"column:case_sign;type:varchar(255)"`
	InputData      string     `gorm:"column:input_data;type:longtext"`
	OutputData     string     `gorm:"column:output_data;type:longtext"`
	ExpectedOutput string     `gorm:"column:expected_output;type:longtext"`
	InputFilePath  string     `gorm:"column:input_file_path;type:varchar(500)"`
	InputFileSize  int64      `gorm:"column:input_file_size;default:0"`
	OutputFilePath string     `gorm:"column:output_file_path;type:varchar(500)"`
	OutputFileSize int64      `gorm:"column:output_file_size;default:0"`
	MaxTime        float64    `gorm:"column:max_time;type:decimal(10,2);default:0.00"`
	MaxMemory      float64    `gorm:"column:max_memory;type:decimal(10,2);default:0.00"`
	IsSample       bool       `gorm:"column:is_sample;default:false"`
	Score          float64    `gorm:"column:score;type:decimal(10,2);default:0.00"`
	Status         string     `gorm:"column:status;type:varchar(32)"`
	Message        string     `gorm:"column:message;type:text"`
	ExitCode       int        `gorm:"column:exit_code;default:0"`
	Deleted        bool       `gorm:"column:deleted;default:false"`
	CreateTime     *time.Time `gorm:"column:create_time"`
	CreateUser     string     `gorm:"column:create_user;type:varchar(32)"`
	UpdateTime     *time.Time `gorm:"column:update_time"`
	UpdateUser     string     `gorm:"column:update_user;type:varchar(32)"`
}

func (DataJudgeCase) TableName() string {
	return "data_judge_case"
}
